the struct may have padding fields and we cannot
rely on it initializing correct fields like this

--- a/netifaces.c
+++ b/netifaces.c
@@ -1800,17 +1800,17 @@ gateways (PyObject *self)
     do {
       struct sockaddr_nl sanl_from;
       struct iovec iov = { msgbuf, bufsize };
-      struct msghdr msghdr = {
-        &sanl_from,
-        sizeof(sanl_from),
-        &iov,
-        1,
-        NULL,
-        0,
-        0
-      };
+      struct msghdr msghdr = {0};
       int nllen;
 
+      msghdr.msg_name = &sanl_from;
+      msghdr.msg_namelen = sizeof(sanl_from);
+      msghdr.msg_iov = &iov;
+      msghdr.msg_iovlen = 1;
+      msghdr.msg_control = NULL;
+      msghdr.msg_controllen = 0;
+      msghdr.msg_flags = 0;
+
       ret = recvmsg (s, &msghdr, 0);
 
       if (msghdr.msg_flags & MSG_TRUNC) {
